### PR TITLE
feat: settings page

### PR DESCRIPTION
**Pull request by simboonlong**
- to have settings page with 2 links
- 1 link is address book
- 1 link is address resolver

### user story
- https://www.pivotaltracker.com/story/show/175569094

### CODE REVIEW

File: src/components/AddressBookContainer/AddressBookContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { usePersistedConfigFile } from "./../../common/hook/usePersistedConfigFile";
+import { Link } from "react-router-dom";
+import React, { FunctionComponent } from "react";
+import { NavigationBar } from "../NavigationBar";
+
+export const AddressBookContainer: FunctionComponent = () => {
+  const { configFile } = usePersistedConfigFile();
+  return (
+    <>
+      <NavigationBar />
```
Comment (isaackps) [inline src/components/AddressBookContainer/AddressBookContainer.tsx:11]: Do u think the nav bar can be in the router level?

**Review by isaackps — CHANGES_REQUESTED** [review]

File: src/components/AddressBookContainer/AddressBookContainer.tsx (hunk 1 above)
Comment (simboonlong) [reply src/components/AddressBookContainer/AddressBookContainer.tsx:11]: i think i tried b4 like cannot due to how this project was been setup (the part on various form routes logic), i can try again see how let u know again

Comment (simboonlong) [reply src/components/AddressBookContainer/AddressBookContainer.tsx]: ok looks like can 👍

File: src/components/AddressResolverContainer/AddressResolverContainer.tsx [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
     <>
-      <NavigationBar />
-      <AddressResolver />
+      <div className="container py-8">
+        <Link to="/settings">
+          <BackArrow />
+        </Link>
+        <AddressResolver />
+      </div>
     </>
```
Comment (isaackps) [inline src/components/AddressResolverContainer/AddressResolverContainer.tsx:14]: can remove the <> </> liao

File: src/components/AddressBookContainer/AddressBookContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <>
+      <div className="container py-8">
+        <Link to="/settings">
+          <BackArrow />
+        </Link>
+        <h3>Settings: Address Book</h3>
+        <AddressBook network={configFile?.network} />
+      </div>
+    </>
```
Comment (isaackps) [inline src/components/AddressBookContainer/AddressBookContainer.tsx:17]: can remove the <> </> liao.

File: src/components/DynamicFormContainer/DynamicFormContainer.tsx [hunk 4]
```diff
@@ -1,26 +1,16 @@
...
+  const { config } = useConfigContext();
 
   if (!config) {
     return <Redirect to="/" />;
   }
   return (
     <>
-      <NavigationBar logout={logout} />
       <DynamicFormLayout />
     </>
```
Comment (isaackps) [inline src/components/DynamicFormContainer/DynamicFormContainer.tsx:15]: same thing, can remove the <> </>

File: src/components/FormSelectionContainer/FormSelectionContainer.tsx [hunk 5]
```diff
@@ -1,26 +1,17 @@
...
 
   if (!config) {
     return <Redirect to="/" />;
   }
+
   return (
     <>
-      <NavigationBar logout={logout} />
       <FormSelection config={config} />
     </>
```
Comment (isaackps) [inline src/components/FormSelectionContainer/FormSelectionContainer.tsx:16]: same thing as above

File: src/components/Home/Home.tsx [hunk 6]
```diff
@@ -13,10 +12,5 @@ export const HomeContainer: FunctionComponent = () => {
   // If wallet has been decrypted, redirect to forms
   if (config) return <Redirect to="/forms-selection" />;
 
-  return (
-    <>
-      <NavigationBar />
-      {configFile ? <WalletDecryptionContainer /> : <ConfigFileDropZoneContainer />}
-    </>
-  );
+  return <>{configFile ? <WalletDecryptionContainer /> : <ConfigFileDropZoneContainer />}</>;
```
Comment (isaackps) [inline src/components/Home/Home.tsx:15]: same as above

File: src/components/PublishContainer/PublishPage/PublishPage.tsx [hunk 7]
```diff
@@ -31,7 +30,6 @@ export const PublishPage: FunctionComponent<PublishPage> = ({ config }) => {
 
   return (
     <>
```
Comment (isaackps) [inline src/components/PublishContainer/PublishPage/PublishPage.tsx:32]: same thing, can remove this

File: src/components/SettingsContainer/SettingsContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+          <br />
+          <Link to="/settings/address-resolver">
+            <TileInfo
+              title="Address Book Resolver"
+              description="Set up and add third party’s endpoint to resolve addresses’ identity"
+            />
+          </Link>
+        </div>
+      </div>
+    </>
```
Comment (isaackps) [inline src/components/SettingsContainer/SettingsContainer.tsx:23]: same can remove it liao

File: src/Router.tsx [hunk 9]
```diff
@@ -2,23 +2,43 @@ import { NetworkBar, Overlay } from "@govtechsg/tradetrust-ui-components";
...
   return (
     <>
       <NetworkBar network={configFile?.network}>
         You are currently on <span className="capitalize">{configFile?.network}</span> network. To
         change it, please upload a new config file.
       </NetworkBar>
-      <main className="bg-blue-300">
-        <BrowserRouter>
+      <BrowserRouter>
+        <NavigationBar {...navigationBarProps} />
```
Comment (isaackps) [inline src/Router.tsx:40]: y create another const to handle the navigationBarProps? can do something like this in the props?
```
logout={config ? logout : undefined}
or
config && logout={logout}
```

**Review by isaackps — CHANGES_REQUESTED** [review]

File: src/Router.tsx (hunk 9 above)
Comment (simboonlong) [reply src/Router.tsx:40]: right... wat was i thinking lol

**Review by isaackps — APPROVED** [review]
LGTM!

**Pull request merged**